HLTrigger/HLTcore/interface/TriggerSummaryProducerAOD.h

```diff
@@ -61,16 +64,34 @@ namespace edm {
 // class declaration
 //
 
-class TriggerSummaryProducerAOD : public edm::EDProducer {
+/// GlobalCache
+struct InputTagHash {
+  std::size_t operator()(const edm::InputTag& inputTag) const {
+    std::hash<std::string> Hash;
+    return Hash(inputTag.encode());
```

Comment:
This should work, but might not be overly fast. An alternative would be to hash the three tags and then xor the results. That would avoid having to do string manipulations and memory allocations.